lib/client.dart

```diff
@@ -1,59 +1,141 @@
 import 'dart:async';
 import 'dart:convert';
+
+import 'package:gql/execution.dart' as exec;
+import 'package:gql/language.dart';
+import 'package:gql/link.dart' as gql_link;
 import 'package:http/http.dart' as http;
 import 'package:json_annotation/json_annotation.dart';
+
 import './schema/graphql_error.dart';
 import './schema/graphql_query.dart';
 
-class _DefaultHttpJsonClient extends http.BaseClient {
-  http.Client _httpClient = http.Client();
+/// Simple implementation of an HttpLink
+class SimpleHttpJsonLink implements gql_link.Link {
+  /// Endpoint of a GraphQL service
+  final String graphQLEndpoint;
+
+  http.Client _httpClient;
+
+  /// Construct the Link
+  SimpleHttpJsonLink(this.graphQLEndpoint, {http.Client httpClient}) {
+    this._httpClient = httpClient ?? http.Client();
+  }
+
+  Future<exec.Response> _post(exec.Request request) async {
+    final dataResponse = await _httpClient.post(
```

Comment:
Instead of always calling the `POST` on the client WDYT about making this configurable (it can default to post)? I know that POST is the most usual method used, however, there are cases that use GET instead. (especially because we are trying to decouple the transmission protocol from the GraphQL itself with those changes)

Reply:
I think this `SimpleHttpJsonLink` is just to easy for user that does not want to spend time implement their own Link.
But, I think it could be kept as simple as it is right now, and for more configurable one, the user should implement their own.
Also, I think it could be nice a little _how to_ about implement your own `Link`

Reply:
I agree about it, however, I believe that the most "common" Links could be shipped with the package (just like Apollo does with Http, Auth, etc.. Links), this would eliminate the friction of implementing a whole new Link for just changing small settings (like Method, or Headers) and will make users able to get all goodies of the default implementation

Reply:
Yes, once Dart has the basic set of Links, user will mostly use those and extend them only when some rare special conditions are met.

Of course, we are not there yet. Even `package:graphql` implements only the must-have links.

Reply:
@leoiacovini I am all for supporting `GET` as well. But just today I read how `apollo-link-http` does it and it appears to be a bit too involved task to implement right now.